src/nvim/fold.c

```diff
@@ -1835,10 +1835,11 @@ char_u *get_foldtext(win_T *wp, linenr_T lnum, linenr_T lnume,
       curwin = wp;
       curbuf = wp->w_buffer;
```

Comment:
Why not removing the `curwin` dance here too ?

Reply:
After a bit of testing, I figured it breaks some tests in `diff_spec`.

Reply:
yep that's why I had to rollback, otherwise foldtext runs on the wrong window.